pyteal/compiler/optimizer/optimizer.py

```diff
@@ -0,0 +1,73 @@
+from typing import List, cast
+
+from pyteal.ir.tealcomponent import TealComponent
+from pyteal.ir.tealop import TealOp
+from pyteal.ir.ops import Op
+
+
+class OptimizeOptions:
+    """An object which specifies optimizations to be performed."""
+
+    use_slot_to_stack: bool
+    iterate_optimizations: bool
+
+    def __init__(
+        self, *, useSlotToStack: bool = False, iterateOptimizations: bool = False
+    ):
+        """Create a new OptimizeOptions object.
+        Args:
+            useSlotToStack (optional): todo
+            iterateOptimizations (optional): todo
+        """
+        self.use_slot_to_stack = useSlotToStack
+        self.iterate_optimizations = iterateOptimizations
+
+
+def has_load_dependencies(teal: List[TealComponent], slot: int, pos: int):
+    for i, op in enumerate(teal):
+        if i == pos:
+            continue
+
+        if type(op) == TealOp and op.op == Op.load and op.args[0] == slot:
+            return True
+
+    return False
+
+
+def apply_slot_to_stack(teal: List[TealComponent]) -> List[TealComponent]:
```

Comment:
@algoidurovic Optional - Feel welcomed to resolve as is:  I'd optionally consider prefixing `apply_slot_to_stack` and `has_load_dependencies` with `_` to denote private methods.

From what I can tell, these methods are _not_ intended to be generally available.